openapi-cli/src/main/java/io/ballerina/openapi/cmd/OpenAPIDiagnostic.java

```diff
@@ -36,14 +36,18 @@ public class OpenAPIDiagnostic extends Diagnostic {
     private final DiagnosticInfo diagnosticInfo;
     private final Location location;
     private final List<DiagnosticProperty<?>> properties;
-    private final String message;
+    private  String message = null;
 
     public OpenAPIDiagnostic(DiagnosticInfo diagnosticInfo, Location location, List<DiagnosticProperty<?>> properties,
                              Object[] args) {
         this.diagnosticInfo = diagnosticInfo;
         this.location = location;
         this.properties = properties;
-        this.message = MessageFormat.format(diagnosticInfo.messageFormat(), args);
+        try {
+            this.message = MessageFormat.format(diagnosticInfo.messageFormat(), args);
```

Comment:
When does this return `IllegalArgumentException`?

Reply:
This was returned when the given JSON has parser issues, we need to write diagnostic with actual parser error message in that case if we directly pass that message into `format` it will return this `IllegalArgumentException` exception due to the passer error message `contains` several brackets that break `.format`.  That is why do a catch there. ex: https://github.com/ballerina-platform/openapi-tools/pull/1739/files#diff-13ca321a2e754d015db6bcb861e3a19172c1d72026581ae53ab242cfe4e4aacaR90

Reply:
I would prefer to hide this Java parser error and return a general error like `Error occurred while parsing the given example file`.

Reply:
The purpose of adding the parser issue is to give the user awareness about which area issue occurred, then needs to be fixed in the Java sample. on the other way around  as you suggested we can let the user with the given example have some parser issues and they can check those

Reply:
IMO giving the entire java parser error does not look nice

Reply:
Please review https://github.com/ballerina-platform/openapi-tools/pull/1739/commits/0696208f03ba08b303f5041d78356a7eb257af73